src/composables/Project.js

```diff
@@ -1,10 +1,72 @@
 import { randomHexString } from 'src/composables/Random';
 
+export const PROJECT_STORAGE_KEY = 'projects';
+
+/**
+ * @typedef {Object} Project
+ * @property {string} id project id
+ */
+
 /**
  * Create a project with generated id.
  *
- * @returns {{Object}} Project object with generated id.
+ * @returns {Project} Project object with generated id.
  */
-export function createProject() {
+export function createProjectTemplate() {
   return { id: `project-${randomHexString(8)}` };
 }
+
+/**
+ * Get a map of all projects.
+ *
+ * @return {Record<string, Project>} a map of all projects
+ */
+export function getProjects() {
+  const lsProjects = localStorage.getItem(PROJECT_STORAGE_KEY);
+  return JSON.parse(lsProjects || '{}');
+}
+
+/**
+ * Get one project by its ID.
+ *
+ * @param {string} projectId
+ * @returns {Project}
+ */
+export function getProjectById(projectId) {
+  const projects = getProjects();
+  return projects[projectId];
+}
+
+/**
+ * Save the project state.
+ *
+ * @param {Project} project
+ */
+export function saveProject(project) {
+  const projects = getProjects();
+  projects[project.id] = project;
+  localStorage.setItem(PROJECT_STORAGE_KEY, JSON.stringify(projects));
+}
+
+/**
+ * Clear one project by ID.
+ *
+ * @param {string} projectId
+ * @return {boolean} if the project was deleted or not
+ */
+export function deleteProjectById(projectId) {
+  const projects = getProjects();
+  if (projects[projectId]) {
+    delete projects[projectId];
+    localStorage.setItem(PROJECT_STORAGE_KEY, JSON.stringify(projects));
+    return true;
+  }
+  return false;
```

Comment:
Why is it needed to return something for this function ?

Reply:
Conventions for these kind of functions, not needed here yet. I'll remove it.